apps/activity/views.py

```diff
@@ -1,5 +1,185 @@
-from rest_framework import viewsets
+from datetime import timedelta
+from django.shortcuts import get_object_or_404
 
+from django.utils import timezone
+from drf_yasg.utils import swagger_auto_schema, no_body
+from rest_framework import filters
+from rest_framework import viewsets, status
+from rest_framework.decorators import action
+from rest_framework.response import Response
 
-class ActivityViewSet(viewsets.ModelViewSet):
-    pass
+from apps.tasks.models import Task
+from apps.tasks.serializers import GetTaskSerializer
+from apps.activity.models import Plan, ActivitySleep, ActivityWater, ActivityFood, ActivityStep
+from apps.activity.serializers import PlanSerializer, SleepSerializer, WaterSerializer, CreateFoodSerializer,\
+    CreateStepsSerializer, GetAllStepsSerializer, GetAllCaloriesSerializer
+
+
+class PlanViewSet(viewsets.ModelViewSet):
+    queryset = Plan.objects.all()
+    filter_backends = [filters.OrderingFilter]
+    ordering = ['id']
+    serializer_class = PlanSerializer
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        if getattr(self, "swagger_fake_view", False):
+            return Plan.objects.none()
+        if self.action == "tasks":
+            return self.request.user.plan.tasks.all().order_by("id")
+
+        return queryset
+
+    def perform_create(self, serializer):
+        plan = serializer.save()
+        self.request.user.plan = plan
+        self.request.user.save()
+
+    @action(methods=['patch'], detail=False, serializer_class=PlanSerializer, url_path="change")
+    def change(self, request, *args, **kwargs):
+        plan = Plan.objects.get(id=request.user.plan.id)
+        plan.steps = request.data["steps"]
+        plan.calories = request.data["calories"]
+        plan.sleep = request.data["sleep"]
+        plan.water = request.data["water"]
+        plan.save()
+        return Response({"success": True}, status.HTTP_200_OK)
+
+    @action(methods=['post'], detail=False, serializer_class=SleepSerializer, url_path="sleep")
+    def sleep(self, request, *args, **kwargs):
+        sleep = request.data["sleep"]
+        plan = self.request.user.plan
+        ActivitySleep.objects.create(sleep=sleep, plan=plan)
+        return Response({"success": True}, status.HTTP_201_CREATED)
+
+    @action(methods=['post'], detail=False, serializer_class=WaterSerializer, url_path="water")
+    def water(self, request, *args, **kwargs):
+        sleep = request.data["water"]
+        plan = self.request.user.plan
+        ActivityWater.objects.create(water=sleep, plan=plan)
+        return Response({"success": True}, status.HTTP_201_CREATED)
```

Comment:
Nu inteleg de ce la toate views se returneaza succes: True, nui corect asa, de exemplu in cazul dat se creaza o instanta de ActivityWater dc nu o returnez pe ea in response.